test/functional/Java8andUp/src/org/openj9/test/string/StringStreams.java

```diff
@@ -0,0 +1,161 @@
+			expectedCharacteristics |= Spliterator.IMMUTABLE; 
+		}
+		int characteristics = s.characteristics();
+		assertTrue((characteristics & expectedCharacteristics) == expectedCharacteristics,
+				"codePoints() spliterator characteristics 0x" + Integer.toHexString(characteristics)  //$NON-NLS-1$
+				+ "\ndid not have expected characteristics 0x" + Integer.toHexString(expectedCharacteristics)); //$NON-NLS-1$
+	}
+
+	@Test
+	public void testStringCharsSpliterator() {
+		Spliterator.OfInt s = DIGITS.codePoints().spliterator();
+		final ArrayList<Integer> actualIntegers = new ArrayList<>();
+		s.forEachRemaining((IntConsumer) i -> actualIntegers.add(Integer.valueOf(i)));
+		for (int c = 0; c < DIGITS.length(); ++c) {
+			assertEquals(actualIntegers.get(c), Integer.valueOf(DIGITS.codePointAt(c)), 
+					"Wrong value at position " + c); //$NON-NLS-1$
+		}
+	}
+
+	@Test
+	public void testStringCodePointsSpliterator() {
+		Spliterator.OfInt s = DIGITS.chars().spliterator();
+		final ArrayList<Integer> actualIntegers = new ArrayList<>();
+		s.forEachRemaining((IntConsumer) i -> actualIntegers.add(Integer.valueOf(i)));
+		checkDigitsOrdered(DIGITS, actualIntegers);
+	}
+
+	@Test
+	public void testStringCharsSpliteratorSingleChar() {
+		Spliterator.OfInt s = SINGLECHAR.chars().spliterator();
+		final ArrayList<Integer> actualIntegers = new ArrayList<>();
+		s.forEachRemaining((IntConsumer) i -> actualIntegers.add(Integer.valueOf(i)));
+		checkDigitsOrdered(SINGLECHAR, actualIntegers);
+	}
+
+	@Test
+	public void testStringCharsSpliteratorEmpty() {
+		Spliterator.OfInt s = EMPTY.chars().spliterator();
+		final ArrayList<Integer> actualIntegers = new ArrayList<>();
+		s.forEachRemaining((IntConsumer) i -> actualIntegers.add(Integer.valueOf(i)));
+		checkDigitsOrdered(EMPTY, actualIntegers);
+	}
+
+	@Test
+	public void testStringCharsSpliteratorSplitting() {
+		OfInt s = DIGITS.chars().spliterator();
+		ArrayList<OfInt> sArray = new ArrayList<>();
+		while (Objects.nonNull(s)) {
+			sArray.add(s);
+			s = s.trySplit();
+		}
+		final HashSet<Integer> actualIntegers = new HashSet<>();
+		for (OfInt e: sArray) {
+			e.forEachRemaining((IntConsumer) i -> actualIntegers.add(Integer.valueOf(i)));
+		}
+		checkDigits(actualIntegers);
+	}
+
+	/**
+	 * Consume a bit of the spliterator, then split it.
+	 */
+	@Test
+	public void testStringCharsSpliteratorSplittingAfterIterating() {
+		OfInt s = DIGITS.chars().spliterator();
+		ArrayList<OfInt> sArray = new ArrayList<Spliterator.OfInt>();
+		final HashSet<Integer> actualIntegers = new HashSet<>();
+		IntConsumer consumer = (IntConsumer) i -> actualIntegers.add(Integer.valueOf(i));
+		while (Objects.nonNull(s)) {
+			s.tryAdvance(consumer);
+			sArray.add(s);
+			s = s.trySplit();
+		}
+		for (OfInt e: sArray) {
+			e.forEachRemaining(consumer);
+		}
+		checkDigits(actualIntegers);
+	}
+
+	private void checkDigitsOrdered(String expectedString, ArrayList<Integer> actualIntegers) {
```

Comment:
This method can be `static`; the second parameter should be more general: `List<Integer>`.